### PR TITLE
Add the option to update trackMetadata

### PR DESCRIPTION
**Pull request by HybridFox**
Works for android & iOS. Handy for icecast / shoutcast / etc... audio streams.

#194 #93 #192

### CODE REVIEW

**Comment by leonhh** [comment]
Any chance this can get merged soon?

**Comment by HybridFox** [comment]
Is there a possibility for an update on this?

**Comment by Guichaguri** [comment]
I'm waiting for a review on the iOS part from @dcvz, once he gives a green light, we can merge it.

File: ios/RNTrackPlayer/Logic/MediaWrapper.swift [hunk 1]
```diff
@@ -112,6 +112,14 @@ class MediaWrapper: AudioPlayerDelegate {
         }
     }
     
+    func updateMetadata(_ item: Track) {
+        MPNowPlayingInfoCenter.default().ap_update(
+            with: item,
+            duration: 100,
```
Comment (dcvz) [inline ios/RNTrackPlayer/Logic/MediaWrapper.swift]: we should have track duration from the item, right? perhaps even progression can be inferred here.

**Comment by Guichaguri** [comment]
You're dynamically sending metadata updates, don't you think it would be better if you updated the current track object instead?

I was thinking about something like `updateTrack(id, obj)`, which allows you to update only metadata-related properties. That way, it not only allows you to update metadata from other tracks in the queue, but also keeps the metadata changes through any playback updates.

For instance, if you have used `updateMetadata` and the track is stopped and started again, you'll have to use `updateMetadata` again. The same can happen with any other playback update.

**Pull request merged**